pkg/fanal/analyzer/pkg/dpkg/copyright_test.go

```diff
@@ -0,0 +1,150 @@
+package dpkg
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDpkgLicensesAnalyzer_Analyze(t *testing.T) {
+	tests := []struct {
+		name              string
+		copyrightFilePath string
+		wantLicense       *analyzer.AnalysisResult
+	}{
+		{
+			name:              "happy path. There are 'License:' pattern and licenseclassifier",
+			copyrightFilePath: "testdata/copyrightFiles/usr/share/doc/zlib1g/copyright",
+			wantLicense: &analyzer.AnalysisResult{
+				CustomResources: []types.CustomResource{
+					{
+						Type:     LicenseAdder,
+						FilePath: "zlib1g",
+						Data:     "Zlib",
+					},
+				},
+			},
+		},
+		{
+			name:              "happy path. There is Common license",
+			copyrightFilePath: "testdata/copyrightFiles/usr/share/doc/adduser/copyright",
+			wantLicense: &analyzer.AnalysisResult{
+				CustomResources: []types.CustomResource{
+					{
+						Type:     LicenseAdder,
+						FilePath: "adduser",
+						Data:     "GPL-2, GPL-2.0",
+					},
+				},
+			},
+		},
+		{
+			name:              "happy path. There are Common license, 'License:' pattern and licenseclassifier",
+			copyrightFilePath: "testdata/copyrightFiles/usr/share/doc/apt/copyright",
+			wantLicense: &analyzer.AnalysisResult{
+				CustomResources: []types.CustomResource{
+					{
+						Type:     LicenseAdder,
+						FilePath: "apt",
+						Data:     "GPLv2+, GPL-2, GPL-2.0",
+					},
+				},
+			},
+		},
+		{
+			name:              "happy path. Licenses not found",
+			copyrightFilePath: "testdata/copyrightFiles/usr/share/doc/tzdata/copyright",
+			wantLicense: &analyzer.AnalysisResult{
+				CustomResources: []types.CustomResource{
+					{
+						Type:     LicenseAdder,
+						FilePath: "tzdata",
+						Data:     "Unknown",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			f, err := os.Open(test.copyrightFilePath)
+			if err != nil {
+				t.Error("unable to read test file")
+			}
+
+			input := analyzer.AnalysisInput{
+				Content:  f,
+				FilePath: strings.TrimPrefix(test.copyrightFilePath, "testdata/copyrightFiles/"),
```

Comment:
You can pass a file path regardless of the test file path because we test the logic of package name extraction in `TestDpkgAnalyzer_getPkgNameFromLicenseFilePath`.

```suggestion
				FilePath: "/usr/share/doc/apt/copyright",
```

Reply:
Done